networking_aci/plugins/ml2/drivers/mech_aci/cobra_client.py

```diff
@@ -95,7 +102,7 @@ def __init__(self, hosts, user, password, ssl, verify=False, request_timeout=90)
         self.login()
 
     def login(self):
-        # TODO handle multiple hosts
+        # NOTE: login() is called by _retry(). To avoid deadlocks it should not call any methods that trigger a _retry()
```

Comment:
You mean to say login itself shall not call any function with a `_retry` decorator? Wouldn't we end in a RecursionError then?

Reply:
If the query fails, triggers a login and login would trigger a `_retry()`, then it would wait for the lock that is hold by the same process, thus deadlock.